api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+swagger: '2.0'
+info:
+  title: Nakadi Event Bus API Definition
+  description: |
+    -----------
+    Definitions
+    -----------
+    Nakadi at its core aims at being a generic and content-agnostic event broker with a convenient API.
+    In doing this, Nakadi abstracts away, as much as possible, details of the backing messaging
+    infrastructure. The single currently supported messaging infrastructure is Kafka (Kinesis is planned
+    for the future).
+
+    In Nakadi every Event has an EventType, and a **stream** of Events is exposed for each registered
+    EventType.
+
+    An EventType defines properties relevant for the operation of its associated stream, namely:
+
+    * The **schema** of the Event of this EventType. The schema defines the accepted format of Events of an
+    EventType and will be, if so desired, enforced by Nakadi. Usually Nakadi will respect the schema for
+    the EventTypes in accordance to how an owning Application defines them.
+    **Note:** *Currently the specification of the schema must be pushed into Nakadi on EventType creation;
+    in the future, assuming that Applications will expose the schema for its owned resources, Nakadi might
+    support fetching the schema directly from them.*
+    Nakadi might define globally minimal schema compliance requirements (i.e. some fields that must be
+    present on schemas of all EventTypes).
+
+    * The expected **validation** and **enrichment** procedures upon reception of an Event.
+    (*TODO: define validation and enrichment*)
+
+    * The **ordering** expectations of Events in this stream. (*TODO: define ordering not in terms of
+    partitioning*)
```

Comment:
The notion of "partition" is important element of this API -- and this is fine. 
Some statements about ordering properties of event streams would be helpful -- and it should be related to partitioning.